File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        int n = happiness.size();
        sort(happiness.begin(), happiness.end());
        int i = 0, j = n - 1; 
        long long res = 0;
        while(k > 0){
            int x = happiness[j] - i;
            if(x <= 0) break;
            res += x;
            i++;
            j--;
            k--;
        }
        return res;
    }
};

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        n = len(happiness)
        happiness.sort()
        i = n - 1
        j = 0
        s = 0
        while k > 0:
            x = happiness[i] - j
            if x <= 0:
                break
            else:    
                s += x
            k -= 1
            j += 1
            i -= 1
        return s    
